lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -159,47 +159,47 @@ import qualified Data.Text.Encoding as T
 --
 -- @m | purpose' | cointype' | account' | role | address@
 data Depth
-    = RootK | PurposeK | CoinTypeK | AccountK | RoleK | AddressK
+    = RootK | PurposeK | CoinTypeK | AccountK | RoleK | AddressK | ScriptK
 
--- | Marker for addresses type engaged. We want to handle three cases here.
--- The first two are pertinent to UTxO accounting
--- and the last one handles rewards from participation in staking.
+-- | Marker for addresses type engaged. We want to handle four cases here.
+-- The first two are pertinent to UTxO accounting,
+-- next handles rewards from participation in staking
+-- the last one is used for getting verification keys used in scripts.
 -- (a) external chain is used for addresses that are part of the 'advertised'
 --     targets of a given transaction
 -- (b) internal change is for addresses used to handle the change of a
 --     the transaction within a given wallet
 -- (c) the addresses for a reward (chimeric) account
---
--- FIXME: rename this to 'Role' or 'HDRole'
-data AccountingStyle
+-- (d) used for keys used inside scripts
+data Role
     = UtxoExternal
     | UtxoInternal
     | MutableAccount
```

Comment:
Now that Jormungandr is gone, we will want to rename that one to `Stake` probably. But that's for later.

Reply:
I will do in one of the next PRs if you don't mind. This one got very big already